tests/integration/components/make-decision-dropdown/component-collection-test.ts

```diff
@@ -0,0 +1,168 @@
+        const store = testContext.owner.lookup('service:store');
+        const collectionSubmission = await store.findRecord('collection-submission', submissionId);
+        testContext.set('collectionSubmission', collectionSubmission);
+    }
+
+    // Dynamic creating of all test cases based on the review states
+    for (const reviewsState of (Object.keys(collectionSubmissionTestCases) as CollectionSubmissionReviewStates[])) {
+        test(`shows actions for collection submission state: ${reviewsState}`,
+            async function(this: TestContext, assert) {
+                await buildCollectionScenarios(this, reviewsState);
+
+                const testCase = collectionSubmissionTestCases[reviewsState];
+
+                // Given the MakeDecisionDropdown is rendered
+                await render(hbs`<MakeDecisionDropdown @collectionSubmission={{this.collectionSubmission}} />`);
+
+                // When the dropdown button is clicked
+                await click('[data-test-moderation-dropdown-button]');
+
+                testCase.actions.forEach((actionTrigger: CollectionSubmissionActionTrigger, $index: number) => {
+                    // Then the "actionTrigger" checkbox option is available
+                    assert.dom(`[data-test-moderation-dropdown-decision-checkbox=${actionTrigger}]`)
+                        .isVisible(`'${actionTrigger}' checkbox option is visible`);
+
+                    // And the "actionTrigger" checkbox option is not checked
+                    assert.dom(`[data-test-moderation-dropdown-decision-checkbox=${actionTrigger}]`)
+                        .isNotChecked(`'${actionTrigger}' checkbox option is not checked by default`);
+
+                    // And then "actionTrigger" checkbox label is verified
+                    if (testCase.checkboxLabel?.length) {
+                        const labelText = testCase.checkboxLabel[$index];
+                        assert.dom(`[data-test-moderation-dropdown-decision-label=${actionTrigger}]`).hasText(
+                            labelText,
+                            'has the right action trigger description text',
+                        );
+                    }
+
+                    // And the "actionTrigger" checkbox description is verified
+                    if (testCase.checkboxDescription?.length) {
+                        const descriptionText = testCase.checkboxDescription[$index];
+                        assert.dom(`[data-test-moderation-dropdown-decision-description=${actionTrigger}]`).hasText(
+                            descriptionText,
+                            'has the right action trigger description text',
+                        );
+                    }
+                });
+
+
+                if (testCase.actions.length) {
+                    assert.dom('[data-test-moderation-dropdown-action-label]')
+                        .hasText(testCase.commentLabel as string,
```

Comment:
You can also use the TypeScript non-null assertion here like:
`.hasText(testCase.commentLabel!,`
I don't think this is a necessary change, but I just prefer it cause it's a little shorter to write and the intention feels a bit more clear  (ie: this _will_ be truthy vs. this is recast as a string)